crates/cast/bin/cmd/wallet/mod.rs

```diff
@@ -273,12 +276,13 @@ impl WalletSubcommands {
                 println!("0x{sig}");
             }
             WalletSubcommands::Verify { message, signature, address } => {
-                let recovered_address =
-                    signature.recover_address_from_prehash(&B256::from_str(&message)?)?;
-                if recovered_address == address {
-                    println!("Validation succeeded. Address {address} signed this message.");
-                } else {
-                    println!("Validation failed. Address {address} did not sign this message.");
+                match signature.verify(Self::hex_str_to_bytes(&message)?, address.to_ethers()) {
```

Comment:
We should not be using the old ethers type here. We should actually use `recover_address_from_msg` from the alloy `Signature` type.